service.subtitles.napiprojekt/service.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*- 
+
+import os
+import sys
+from urllib.parse import unquote, quote_plus
+import unicodedata
+from hashlib import md5
+
+import xbmc
+import xbmcvfs
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+
+__addon__ = xbmcaddon.Addon()
+__scriptid__ = __addon__.getAddonInfo('id')
+__scriptname__ = __addon__.getAddonInfo('name')
+__version__ = __addon__.getAddonInfo('version')
+__language__ = __addon__.getLocalizedString
+
+__cwd__ = xbmcvfs.translatePath(__addon__.getAddonInfo('path'))
+__profile__ = xbmcvfs.translatePath(__addon__.getAddonInfo('profile'))
+__resource__ = xbmcvfs.translatePath(os.path.join(__cwd__, 'resources', 'lib'))
+__temp__ = xbmcvfs.translatePath(os.path.join(__profile__, 'temp', ''))
+
+sys.path.append(__resource__)
+
+from NapiProjekt import NapiProjektHelper, log
+
+
+def timeout(func, args=(), kwargs={}, timeout_duration=10, default=None):
+    import threading
+
+    class InterruptableThread(threading.Thread):
+        def __init__(self):
+            threading.Thread.__init__(self)
+            self.result = "000000000000"
+
+        def run(self):
+            self.result = func(*args, **kwargs)
+
+    it = InterruptableThread()
+    it.start()
+    it.join(timeout_duration)
+    if it.is_alive():
+        return it.result
+    else:
+        return it.result
+
+
+def get_filehash(path, rar):
+    d = md5()
+    qpath = quote_plus(path, safe='')
+    if rar:
+        path = """rar://""" + qpath + '/'
+        for file in xbmcvfs.listdir(path)[1]:
+            if file.lower().endswith(('.avi', '.mkv', '.mp4')):
+                path = path + file
+                break
+
+    file = xbmcvfs.File(path, "rb")
```

Comment:
with statement?

with xbmcvfs.... as f:
    f.do_something,()

Reply:
Why? it is a matter of tasted, I prefer the code to be expressive (without trying to guess what is this local var `f`).

Reply:
The point is the with statement that will ensure the handler is always closed, not the variable name. Although using file as a variable is also a bad practice since file is a python builtin function.

Reply:
Ha, thanx for pointing it out, I missed the `with` statement :]